### PR TITLE
add filter for CleanCss

### PR DESCRIPTION
**Pull request by mrothauer**
new filter for https://github.com/clean-css/clean-css
feel free to adapt the code if you want

### CODE REVIEW

File: src/Filter/CleanCss.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * Allows you to filter Css files through CleanCss. You need to install CleanCssCli with composer.
+ */
+class CleanCss extends AssetFilter
+{
+
+    /**
+     * Settings for CleanCss
+     *
+     * - `node` Path to nodejs on your machine
+     * - `node_path` The path to the node_modules directory where uglify is installed.
```
Comment (markstory) [inline src/Filter/CleanCss.php]:
```suggestion
     * - `node_path` The path to the node_modules directory where cleancss is installed.
```

File: src/Filter/CleanCss.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+     *
+     * @param  string $target   Name of the file being generated.
+     * @param  string $content The uncompressed contents for $filename.
+     * @return string Compressed contents.
+     */
+    public function output($target, $content)
+    {
+        $tmpFile = tempnam(sys_get_temp_dir(), 'CLEANCSS');
+        file_put_contents($tmpFile, $content);
+        $cmd = $this->_settings['node'] . ' ' . $this->_settings['cleancss'] . $this->_settings['options'] . ' ' . escapeshellarg($tmpFile);
```
Comment (markstory) [inline src/Filter/CleanCss.php]:
```suggestion
        $cmd = $this->_settings['node'] . ' ' .
            $this->_settings['cleancss'] . $this->_settings['options'] . ' ' .
            escapeshellarg($tmpFile);
```

**Review by markstory — COMMENTED** [review]
Any chance there could be a test added for this. I find filter integrations easier to maintain if there is a test covering a small input > output transformation.

**Comment by mrothauer** [comment]
how would I test my filter? it needs cleancss (node module) installed. I wanted to see how it's done by another node filter (uglifyjs) - but there is no test.
should I create a package.json in tests/test_files?
or should I mock it? if so, please tell me how you want it to be implemented

**Comment by markstory** [comment]
>  I wanted to see how it's done by another node filter (uglifyjs) - but there is no test. should I create a package.json in tests/test_files?

Aw poop, I thought I added tests for the existing node filters, but it looks like I never did do that. Don't worry about the test for now, I'll add one when I get around to doing the others.

**Comment by mrothauer** [comment]
Hi @markstory, when can I expect a new release with my contribution?

**Comment by markstory** [comment]
I think we'd need to skip the new tests as I don't want to leave master in a broken state.

**Pull request merged**